apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -73,7 +109,13 @@ export const pageRouter = router({
         resourceId: z.number().optional(),
       }),
     )
-    .query(async ({ input: { siteId, resourceId } }) => {
+    .query(async ({ ctx, input: { siteId, resourceId } }) => {
+      await validateUserPermissions({
+        userId: ctx.user.id,
+        siteId,
+        action: "read",
+      })
+
```

Comment:
next time with root permissions, we might not be able to do this chekc. this check now works by assuming that `resourceId` is `null` (and hence can check for site wide permissions)